Sources/TuistGenerator/Generator/BuildPhaseGenerator.swift

```diff
@@ -62,14 +62,26 @@ final class BuildPhaseGenerator: BuildPhaseGenerating {
                              pbxTarget: PBXTarget,
                              fileElements: ProjectFileElements,
                              pbxproj: PBXProj,
-                             sourceRootPath _: AbsolutePath) throws {
+                             sourceRootPath: AbsolutePath) throws {
         if let headers = target.headers {
             try generateHeadersBuildPhase(headers: headers,
+                                          includePublicFiles: target.modulemap == nil,
```

Comment:
If I'm not mistaken the criteria here would be to omit the entire build phase if `product == .staticLibrary`,

we can introduce a helper in `TuistCore/Target` along side some of the others there to achieve something along those lines:

```swift
if let headers = target.headers, target.supportsHeadersPhase {
   try generateHeadersBuildPhase( ... )
}
```